Modelica_DeviceDrivers/Resources/Include/MDDSerialPort.h

```diff
@@ -101,7 +102,13 @@ DWORD WINAPI MDD_serialPortReceivingThread(LPVOID p_serial) {
                         if (x > serial->bufferSize) {
                             x = (DWORD)serial->bufferSize;
                         }
-                        ret = ReadFile(serial->hComm, serial->receiveBuffer, x, &serial->receivedBytes, &rdSync);
+                        ret = ReadFile(serial->hComm, &serial->receiveBuffer[idx], x, &serial->receivedBytes, &rdSync);											
+                        idx = idx + serial->receivedBytes;
+                        if(idx == serial->bufferSize)
+                        {
+                          memcpy(serial->completeMessage, serial->receiveBuffer, serial->bufferSize);
+                          idx = 0;
```

Comment:
Ah, I see. You only copy once it is completely filled.

Will merge it soon. Thanks.